vavr/src/test/java/io/vavr/collection/StreamTest.java

```diff
@@ -198,11 +203,73 @@
         return Stream.transpose((Stream<Stream<T>>) rows);
     }
 
+    //fixme: delete, when useIsEqualToInsteadOfIsSameAs() will be eliminated from AbstractValueTest class
     @Override
     protected boolean useIsEqualToInsteadOfIsSameAs() {
         return true;
     }
 
+    @Test
+    @Override
+    public void shouldRemoveNonExistingElement() {
+        final Seq<Integer> t = of(1, 2, 3);
+        assertThat(t.remove(4)).isEqualTo(t);
+        assertNotSame(t, t.remove(4));
+    }
+
+    @Test
+    @Override
+    public void shouldRemoveFirstElementByPredicateNonExisting() {
+        final Seq<Integer> t = of(1, 2, 3);
+        assertThat(t.removeFirst(v -> v == 4)).isEqualTo(t);
+        assertNotSame(t, t.removeFirst(v -> v == 4));
+    }
+
+    @Test
+    @Override
+    public void shouldRemoveLastElementByPredicateNonExisting() {
+        final Seq<Integer> t = of(1, 2, 3);
+        assertThat(t.removeLast(v -> v == 4)).isEqualTo(t);
+        assertNotSame(t, t.removeLast(v -> v == 4));
+    }
+
+    @Test
+    @Override
+    public void shouldNotRemoveAllNonExistingElementsFromNonNil() {
+        final Seq<Integer> t = of(1, 2, 3);
+        assertThat(t.removeAll(of(4, 5))).isEqualTo(t);
+        assertNotSame(t, t.removeAll(of(4, 5)));
+    }
+
+    @Test
+    @Override
+    public void shouldRemoveExistingElements() {
+        final Seq<Integer> seq = of(1, 2, 3);
+        assertThat(seq.removeAll(i -> i == 1)).isEqualTo(of(2, 3));
```

Comment:
`removeAll(Predicate)` is deprecated now - you must use `reject(Predicate)` or add annotation `@SuppressWarnings("deprecation")` to the test method

Reply:
Thank you, I'll change this